aggregator/aggregator.go

```diff
@@ -18,23 +18,24 @@ import (
 )
 
 type TaskProcessor interface {
-	ProcessNewTask(ctx context.Context, event any) (blsagg.TaskMetadata, error)
+	ProcessNewTask(ctx context.Context, log types.Log) (blsagg.TaskMetadata, error)
 	ProcessTaskResponse(ctx context.Context, event TaskResponse) ([32]byte, error)
 	ProcessAggregatedResponse(ctx context.Context, response blsagg.BlsAggregationServiceResponse) error
 }
 
-type Aggregator struct {
+type Aggregator[ResponseType any] struct {
 	logger           logging.Logger
 	serverIpPortAddr string
 	avsWriter        *avsregistry.ChainWriter
+
 	// aggregation related fields
 	blsAggregationService blsagg.BlsAggregationService
 	taskProcessor         TaskProcessor
 	newTaskCreatedLogs    chan types.Log
 }
 
 // NewAggregator creates a new Aggregator with the provided config.
-func NewAggregator(c AggregatorConfig, taskProcessor TaskProcessor, eventHash common.Hash) (*Aggregator, error) {
+func NewAggregator[ResponseType any](c AggregatorConfig, taskProcessor TaskProcessor, eventHash common.Hash) (*Aggregator[ResponseType], error) {
```

Comment:
Doesn't this mean the implementor needs to specify the `ResponseType`? We should check that the generic is resolving correctly.

Reply:
Moved to #634